### PR TITLE
Remove smart union from Generic

### PR DESCRIPTION
**Pull request by oscarbc96**
By default, pydantic tries to validate (and coerce if it can) in the order of the Union. If smart_union is enabled. Pydantic will then check all allowed types before even trying to coerce. This is of course slower, especially if your Union is quite big which is our case.

Our Union is ordered to always coerce to the first possible option, so we don't need the smart union.
Also, this order in combination of the smart_union flag (which was trying to not coerce), was making that booleans like `'true'` and `'false'` not being converted to bools

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,10 @@
 # Change Log
 All notable changes to this project will be documented in this file.
 
+## 0.18.0 - [2022-04-01]
+### Fixes
+- Removes `smart_union` flag from Generic. See https://github.com/Skyscanner/pycfmodel/pull/95
```
Comment (w0rmr1d3r) [inline CHANGELOG.md:6]:
```suggestion
- Removes `smart_union` flag from Generic. See [PR #95](https://github.com/Skyscanner/pycfmodel/pull/95)
```

File: tests/test_generic.py [hunk 2]
```diff
@@ -27,6 +27,10 @@ def test_recursive():
         # ResolvableBoolOrList
         (True, True),
         ([True], [True]),
+        ("true", True),
+        (["true"], [True]),
+        ("True", True),
+        (["True"], [True]),
```
Comment (w0rmr1d3r) [inline tests/test_generic.py:33]: I would add `TRUE` as well

**Review by w0rmr1d3r — COMMENTED** [review]
Good job! Some small additions before is ready

**Review by ignaciobolonio — APPROVED** [review]

**Review by w0rmr1d3r — APPROVED** [review]
Perfect! 🚀

**Pull request merged**